sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -640,11 +640,36 @@ object ScalaJSPluginInternal {
       loadedJSEnv := {
         val log = streams.value.log
         val env = resolvedJSEnv.value
-        val libs =
+        val deps =
           resolvedJSDependencies.value.data ++ scalaJSConfigurationLibs.value
+
+        /* Implement the behavior of commonJSName without having to burn it
+         * inside NodeJSEnv, and hence in the JSEnv API.
+         * Since this matches against NodeJSEnv specifically, it obviously
+         * breaks the OO approach, but oh well ...
+         */
```

Comment:
Ah OK, I see. Indeed `commonJSName` allowed the same `-fastopt.js` to be run in Node.js and in the browser. I do not think this is a use case we need to support.

However, `commonJSName` also allowed facade authors to make the facade agnostic to whether it is run in a module or non-module environment. Examples:

- [Phaser Scala JS](https://github.com/dpitkevics/Phaser-Scala-JS/blob/master/build.sbt)
- [AWS Facade](https://github.com/balshor/aws-sdk-scalajs-facade/blob/master/build.sbt)
- [Raphael Scala.js](https://github.com/yoeluk/raphael-scala-js/blob/master/build.sbt)
- [Scala.js Moment.js](https://github.com/vpavkin/scala-js-momentjs/blob/master/build.sbt)

However, the most prevalent use on GitHub is `React` and `ReactDOM`.

We can easily replace these uses of `commonJSName` with #2797 as follows (at the example of Phaser):
- Library adds `@JSImport("phaser.js", JSImport.Namespace)` on whatever has `@JSGlobal("Phaser")`.
- Users chose appropriate module loading based on JS environment they are using.

I acknowledge that this solves the problem on a different level and is therefore indeed somewhat orthogonal to this PR. However, I still do not understand what your desired end-state with respect to `commonJSName` is (once all this is moved to a separate plugin).

Do you want to:
- Remove it entirely? (Fine with that per-se, community maybe not so much)
- Leave the type test on the JS Env? (Not fine with that :))
- Add an option to the new bundler plugin to chose the behavior? (Probably the best option at this point).